itemManagement/views.py

```diff
@@ -37,15 +40,67 @@ def get_context_data(self, *args, **kwargs):
 class ItemDetailsView(TemplateView):
     def get(self, request, itemId, *args, **kwargs):
         isAdmin = request.user.is_superuser
+        # isAdmin = True
 
         print(f"Item ID requested: {itemId}")
 
         context = Item.objects.get(id=itemId).getItemDetails()
+
         if isAdmin:
             return render(request, 'itemManagement/item_details_admin.html', context=context)
         else:
             return render(request, 'itemManagement/item_details_assistant.html', context=context)
 
+    def post(self, request, itemId, *args, **kwargs):
+
+        item = Item.objects.get(id=itemId)
+
+        item.lastUsed = timezone.now()
+        # TODO: item lastUsed updated if decrement clicked
+
+        isAdmin = True
+
+        # TODO: Admin validation
+
+        # Admin Fields updating fields other than quantities if admin
+        if isAdmin:
+            # --------TEXT FIELDS--------
+            item.title = request.POST.get('itemName', "").strip()
+            item.description = request.POST.get('description', "").strip()
+            item.price = request.POST.get('price', "").strip()
+            item.unit = request.POST.get('unit', "").strip()
+            item.save(update_fields=['title', 'description', 'price', 'unit'])
+            # ---------------------------
+
+            # --------TAGS-------------
+            newTags = request.POST.get('newTags', "").split(',')
+            # clear all tags
+            item.tag_set.all().delete()
+            # add back all new edited tags
+            for newTag in newTags:
+                if newTag != "" or newTag is None:
+                    newTag = Tag.objects.create(name=newTag.strip(), item=item)
+                    newTag.save()
+            # -------------------------
+
+        # ----Quantities in Locations----
+        # Locations: list of all locations that house the current item
+        locations = item.locations.all()
```

Comment:
Instead of getting all locations and then finding the itemstorage that matches, just search for itemStorage and filter for the right item

Reply:
Don't think this makes much of a difference for our purposes. Also when I do add new locations, will I have to manually add entries to `item.locations` and `itemStorage`? Or does `item.locations` come from the `ItemStorage` model?

Reply:
Ya my comment is more for the efficiency of the database and reducing lines of code but its ok. item.locations is related to ItemStorage

Reply:
Made this change